### ISSUE TITLE
Problems fetching at boundaries

### ISSUE THREAD

**Issue by Oscil8** [jimeh/redistat#11]
I've noticed some issues fetching in the last hour of a day, or the last day of a month.  There is a query from 23:00 to the following day at 0:00; or from the beginning of the last day of the month to the beginning of the first day of the next month -- in either case, the query returns the following hour (or day) incorrectly.  I also noticed that a query from x.beginning_of_day to x.end_of_day will always return nothing (when depth is set to day).  Here are the tests I added to finder_spec.rb:

https://gist.github.com/604a6fd34580e472e129

(with corresponding failures)

**Comment by jimeh** [comment]
Thanks, I've noticed some oddities with date selection around boundaries too. To some degree it's expected behavior thanks to how the date selection works, and to some degree it's not cause it's very unintuitive and confusing working with it.

I'll be having a deeper look into it. In the meantime, I recommend you experiment with the Finder::DateSet class and it's spec, as it's the class that the Finder uses to figure out which date sets to select.

**Comment by Oscil8** [comment]
Yes, I took a look at the spec for Finder::DateSet first, but it was even less clear what the correct behavior was; I'll let you know if I figure anything out.

**Comment by jimeh** [comment]
I don't blame you. The date selection code is by far the most complex part of Redistat. I kind of left it behind me and never looked back once I felt confident it worked correctly, but seems I will have to look back now after all... lol
